app/helpers/surveys_helper.rb

```diff
@@ -0,0 +1,138 @@
+module SurveysHelper
+  def surveys_data(type)
+    case type
```

Comment:
I think this could be generalised with some metaprogramming.
If you do `send(type)`, you are actually calling the method named `type`. 
This should prevent this long case statement

Reply:
Oh wow thanks @Levia ! How awesome I will change that now :-)